src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -126,6 +127,9 @@ public Test BuildFrom(IMethodInfo method, Test parentSuite)
                     needCombinatorial = false;
             }
 
+            // This check must be done before CombinatorialAttribute gets added to the builders collection
+            var shouldForceParameterization = builders.Count > 0;
```

Comment:
This entire method would get simpler if we weren't running the CombinatorialAttribute on single tests in case they have parameters. We could decide up front whether the test would be a suite. Perhaps based on whether it has parameters and whether it has builders.